app/views/ai/HackstackAutoPromotion.vue

```diff
@@ -0,0 +1,31 @@
+<script>
+import ModalDynamicPromotion from 'ozaria/site/components/teacher-dashboard/modals/ModalDynamicContent'
+import HackstackPromotionComponent from './HackstackPromotionComponent'
+export default {
+  components: {
+    ModalDynamicPromotion,
+    HackstackPromotionComponent,
+  },
+  methods: {
+    openModalDynamicPromotion () {
+      this.modalDynamicPromotion = true
+    },
+  },
+}
+</script>
+
+<template>
+  <div class="promotion-modal">
+    <ModalDynamicPromotion
+      ref="modal"
+      seen-promotions-property="new-hackstack-promotion-modal"
+    >
+      <template #content>
+        <HackstackPromotionComponent />
+      </template>
```

Comment:
we should handle the close coming from the "learn more" button here
```suggestion
        <HackstackPromotionComponent
          @close="onClose"
        />
```